src/Akka.Persistence.MongoDb/Snapshot/MongoDbGridFSSnapshotStore.cs

```diff
@@ -0,0 +1,356 @@
+            return await act(null, token);
+            
+        using var session = await GetMongoDb().Client.StartSessionAsync(EmptySessionOptions, token);
+        return await session.WithTransactionAsync(act, cancellationToken:token);
+    }
+        
+    private IMongoDatabase GetMongoDb()
+    {
+        if (_mongoDatabase_DoNotUseDirectly is not null)
+            return _mongoDatabase_DoNotUseDirectly;
+        
+        MongoClient client;
+        var setupOption = Context.System.Settings.Setup.Get<MongoDbPersistenceSetup>();
+        if (!setupOption.HasValue || setupOption.Value.SnapshotConnectionSettings == null)
+        {
+            //Default LinqProvider has been changed to LINQ3.LinqProvider can be changed back to LINQ2 in the following way:
+            var connectionString = new MongoUrl(_settings.ConnectionString);
+            var clientSettings = MongoClientSettings.FromUrl(connectionString);
+            clientSettings.LinqProvider = LinqProvider.V2;
+            client = new MongoClient(clientSettings);
+            _mongoDatabase_DoNotUseDirectly = client.GetDatabase(connectionString.DatabaseName);
+            return _mongoDatabase_DoNotUseDirectly;
+        }
+
+        client = new MongoClient(setupOption.Value.SnapshotConnectionSettings);
+        _mongoDatabase_DoNotUseDirectly = client.GetDatabase(setupOption.Value.SnapshotDatabaseName);
+        return _mongoDatabase_DoNotUseDirectly;
+    }
+
+    private GridFSBucket GetGridFSBucket()
+    {
+        var db = GetMongoDb();
+        _snapshotGridFSBucket_DoNotUseDirectly = new GridFSBucket(db, _bucketOptions);
+
+        return _snapshotGridFSBucket_DoNotUseDirectly;
+    }
+
+    private IMongoCollection<GridFSFileInfo> GetFilesCollection()
+    {
+        return GetMongoDb().GetCollection<GridFSFileInfo>(_settings.Collection + ".files");
+    }
+
+    private IMongoCollection<BsonDocument> GetChunksCollection()
+    {
+        return GetMongoDb().GetCollection<BsonDocument>(_settings.Collection + ".chunks");
+    }
+    
+    protected override void PostStop()
+    {
+        // cancel any pending database commands during shutdown
+        _pendingCommandsCancellation.Cancel();
+        _pendingCommandsCancellation.Dispose();
+        base.PostStop();
+    }
+
+    protected override async Task<SelectedSnapshot?> LoadAsync(string persistenceId, SnapshotSelectionCriteria criteria)
+    {
+        using var unitedCts = CreatePerCallCts();
+        var filesCollection = GetFilesCollection();
+        var chunksCollection = GetChunksCollection();
+        
+        var filter = CreateRangeFilter(persistenceId, criteria);
+
+        return await MaybeWithTransaction<SelectedSnapshot?>(async (session, token) =>
+        {
+            var info = await (session is not null 
+                    ? filesCollection.Find(session, filter) 
+                    : filesCollection.Find(filter)) 
+                .SortByDescending(x => x.Metadata[SequenceNrKey])
```

Comment:
Had to write a custom method here for finding only the most recent snapshot, because the sequence number for that snapshot is not known